Tribler/community/tunnel/tunnel_community.py

```diff
@@ -134,26 +139,22 @@ def enabled(self):
     def sendto(self, data, destination):
         if self.check_num_packets(destination, False):
             if TunnelConversion.is_allowed(data):
-                if dispersy.util.is_valid_address(destination):
-                    ip_address = destination[0]
-                else:
+                def on_error(failure):
+                    self.tunnel_logger.error("Can't resolve ip address for hostname %s. Failure: %s",
+                                             destination[0], failure)
+
+                def on_ip_address(ip_address):
+                    self.tunnel_logger.debug("Resolved hostname %s to ip_address %s", destination[0], ip_address)
                     try:
-                        ip_address = socket.gethostbyname(destination[0])
-                        self.tunnel_logger.info("Resolved ip address %s for hostname %s",
-                                                 ip_address,
-                                                 destination[0])
-                    except:
-                        self.tunnel_logger.error("Can't resolve ip address for hostname %s", destination[0])
-
-                try:
-                    self.transport.write(data, (ip_address, destination[1]))
-                except Exception, e:
-                    self.tunnel_logger.error("Failed to write data to transport: %s. Destination: %s",
-                                             e[1],
-                                             repr(destination))
-                    raise
-
-                self.community.increase_bytes_sent(self, len(data))
+                        self.transport.write(data, (ip_address, destination[1]))
+                        self.community.increase_bytes_sent(self, len(data))
+                    except (AttributeError, MessageLengthError) as exception:
+                        self.tunnel_logger.error("Failed to write data to transport: %s. Destination: %r",
```

Comment:
I think it would be more informative to have something like "Failed to write data to transport %s with destination %r. Error was: %r"